src/vs/workbench/contrib/mergeEditor/browser/mergeEditorInput.ts

```diff
@@ -134,14 +134,36 @@ export class MergeEditorInput extends AbstractTextResourceEditorInput implements
 		return this._model;
 	}
 
+	override toUntyped(): IResourceMergeEditorInput {
+		return {
+			input1: { resource: this.input1.uri, label: this.input1.title, description: this.input1.description },
+			input2: { resource: this.input2.uri, label: this.input2.title, description: this.input2.description },
+			base: { resource: this.base },
+			result: { resource: this.result },
+			options: {
+				override: this.typeId
+			}
+		};
+	}
+
 	override matches(otherInput: EditorInput | IUntypedEditorInput): boolean {
-		if (!(otherInput instanceof MergeEditorInput)) {
-			return false;
+		if (this === otherInput) {
+			return true;
 		}
-		return isEqual(this.base, otherInput.base)
-			&& isEqual(this.input1.uri, otherInput.input1.uri)
-			&& isEqual(this.input2.uri, otherInput.input2.uri)
-			&& isEqual(this.result, otherInput.result);
+		if (otherInput instanceof MergeEditorInput) {
+			return isEqual(this.base, otherInput.base)
+				&& isEqual(this.input1.uri, otherInput.input1.uri)
+				&& isEqual(this.input2.uri, otherInput.input2.uri)
+				&& isEqual(this.result, otherInput.result);
+		}
+		if (isResourceMergeEditorInput(otherInput)) {
+			return isEqual(this.base, otherInput.base.resource)
+				&& isEqual(this.input1.uri, otherInput.input1.resource)
+				&& isEqual(this.input2.uri, otherInput.input2.resource)
+				&& isEqual(this.result, otherInput.result.resource);
+		}
```

Comment:
Unsure about how this works. Should we be the same as any merge editor input for the same resources? Should this (already before) take some editor-type into account?

Reply:
Yeah good catch, I forgot the editor ID check because I copied from the default text editor that has no editor ID. Fixed via https://github.com/microsoft/vscode/pull/155039/commits/972decdf237ffc4661ac87fd94b044e4ba47dc65